assembly/onpremises-ide-packaging-war-factory/src/main/java/com/codenvy/onpremises/factory/filter/EscapeIllegalCharactersFactoryURLFilter.java

```diff
@@ -0,0 +1,93 @@
+ * and its suppliers and may be covered by U.S. and Foreign Patents,
+ * patents in process, and are protected by trade secret or copyright law.
+ * Dissemination of this information or reproduction of this material
+ * is strictly forbidden unless prior written permission is obtained
+ * from Codenvy S.A..
+ */
+package com.codenvy.onpremises.factory.filter;
+
+import javax.inject.Singleton;
+import javax.servlet.Filter;
+import javax.servlet.FilterChain;
+import javax.servlet.FilterConfig;
+import javax.servlet.ServletException;
+import javax.servlet.ServletRequest;
+import javax.servlet.ServletResponse;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletRequestWrapper;
+import java.io.IOException;
+import java.util.Map;
+import java.util.regex.Pattern;
+
+/**
+ * Allows to accept the factories by the URLs with old username.
+ *
+ * @author Anton Korneta
+ */
+@Singleton
+public class EscapeIllegalCharactersFactoryURLFilter implements Filter {
+
+    protected static final String USER_NAME = "user";
+
+    private static final Pattern ILLEGAL_USERNAME_CHARACTERS = Pattern.compile("[^a-zA-Z0-9]");
+
+    @Override
+    public void init(FilterConfig filterConfig) throws ServletException {}
+
+    @Override
+    public void doFilter(ServletRequest request, ServletResponse response, FilterChain chain) throws IOException,
+                                                                                                     ServletException {
+        chain.doFilter(new ReplaceUserNameRequestWrapper(((HttpServletRequest)request)), response);
+    }
+
+    @Override
+    public void destroy() {}
+
+    private class ReplaceUserNameRequestWrapper extends HttpServletRequestWrapper {
+
+        public ReplaceUserNameRequestWrapper(HttpServletRequest request) {
+            super(request);
+        }
+
+        @Override
+        public Map<String, String[]> getParameterMap() {
+            final Map<String, String[]> params = super.getParameterMap();
+            final String[] users = params.get(USER_NAME);
+            if (users != null && users.length > 0) {
+                users[0] = replaceUsername(users[0]);
+                params.put(USER_NAME, users);
+            }
+            return params;
+        }
+
+        @Override
+        public String getParameter(String name) {
+            final String parameter = super.getParameter(name);
+            if (name.equals(USER_NAME)) {
+                return replaceUsername(parameter);
+            }
+            return parameter;
+        }
+
+        @Override
+        public String getQueryString() {
+            final String query = super.getQueryString();
+            final String queryParam = "user=";
+            final String username = query.substring(query.indexOf(queryParam) + queryParam.length(),
+                                                    query.length()).split("&")[0];
+            return query.replace(username, replaceUsername(username));
```

Comment:
Does it make sense to have multiple users queries here? Can you guarantee that exactly first of them will be used later?
What will happen with other user queries values? 
If you want to replace 1 user - use replaceFirst